UPPER Case QR Code can be accepted (EXPOSUREAPP-3546)

AndroidMedaGalaxy opened the pull request:
## Description

QR Codes in Upper cases can now be accpted

### Implmentation info
REGEX adjusted and test cases have been added

Please click [here](https://regex101.com/r/RIDKZC/2/) if you want to test more QR code scnearios

<!-- 
Please be brief in describing which issue is solved by your PR or which enhancement it brings. Link related issues!
-->

## Steps to reproduce
step 1. Try the [QR Code generator](https://www.qr-code-generator.com/) and use the following examples.
HTTPS://LOCALHOST/?123456-12345678-1234-4dA7-b166-B86d85475064
https://localhost/?123456-12345678-1234-4DA7-B166-B86D85475ABC

step 2. modify them however you like (Upper case or lowercase letter) 
step 3. Scan them with the phone.
Step 4. You should not see the error for invalid QR

d4rken committed:
Version bump (DEV) (#1498)

BMItr committed:
Improved crash logging & handling (EXPOSUREAPP-2572) (#1475)

* FileLoggerPlus - RX added - preparation

* Buffer config as constants

* avoid empty lines

* Reworked logging to file - wip

* Collecting errors to dao - wip

* Added crash report to app db and created module

* do it

* inject crashReportTree - wip

* use dagger injection

* retrieve short hash via gradle

* removed todo

* Updated database to version 2 for new ReportEntity - old base

* deprecated old engagement - RX not allowed anymore

* use global scope for coroutines in crash report tree - ktlint clean

* added menu item - error reporter (replace later to new structure)

* basic recylcerview and fragment added to menu - wip

* deprecated data class

* Delete CrashModel.kt

* crash report list item to use data binding

* separated Reporteradapter - klint clean

* use separated Adapter in ReporterFragment - cleanup

* Added ErrorReporterViewModel - connected data to list

* dependency injection for crash report, better structure, todo cleanup

* some refactoring, cleanup, ktlint

* adjustments for 1.6.x

* temp action to open crash report fragment

* removed testcode

* updated layout - wip

* connected view & data

* satisfy circle ci checks

* plant reportingTree, testbutton, klint clean

* updated layout for reporter - wip

* better format and layout

* better message handling

* missing icons for errors on debug

* Adjusted error report index

* Removed ugly show button, report clickable

* Added SettingsCrashReportDetailsFragment & corresponding files

* fixed merge inconsistence

* fixed injection overall

* use livedata

* details added to report - wip

* beauty

* share & layout root preparation

* make error report shareable

* removed Dispatcher.IO for example

* Introduce skeleton classes for bugreporting

* Refactoring, adjustments to be able to build again

* Created BugEventEntity, Changed id to Long for more efficiency,  Removed unused events

* refactored ui to deviceForTesters, testmenu etc

* adjusted interfaces & related, better structure

* uses own db, added converter, adjusted modules

* adjusted bugrepo, defaultbugrepo

* shared report, better visualisation

* changed crash db name

* Implemented DefaultBugRepository

* Added device info to BugEvent, BugReporter collects Tag, Added default impl of BugEvent etc

* corrected bug in RollingLogHistory

* removed deprecated crash package

* quickfix - wip

* cleanup

* cleanUp, adjusted simulation button to new structure

* moved bug related layouts to deviceForTesters package

* removed old RX dependenies and related context

* fix build for deviceDebug

* lint clean

* current set of relevant exceptions as before

* attach log to output

* improved output

* Delete obsolet ic_bug in main

* Delete ic_coffee.xml

* changed shortID to shortCommitHash for better transparency

* companion to bottom

* removed string encapsulation

* cleanup unused import and blank line - PR Comment

* typo fixed

* rm empty line

* removed relict module

* layout fix

* adjusted unrelated method call

* fixed package inkonsistence

* Use flow instead of livedata in non-ui

* better fun naming

* Replaced interface with inline listener

* Improved SettingsCrashReportDetailsFragment and SettingsCrashReportViewModel, formatting adjusted

* cleaned converter

* Delete Project.xml

* added log statement to DefaultBugRepository

* adjusted magical number

* Use CommonConverters instead of unique Converters in BugDatabase

* Extension function for the resuce -defaultBugRepository

* removed FormatterCrashReport completly

* changed long to UUID

* satisfy buggy ktlint checks

* Update SettingsCrashReportViewModel.kt

use postvalue

* avoid passing viewmodel

* store appVersionCode as Long for beeing futuresave

* Update SettingsCrashReportViewModel.kt

removed lazy init

* Changed BugDatabase to factory pattern

Co-authored-by: Matthias Urhahn <matthias.urhahn@sap.com>
Co-authored-by: harambasicluka <64483219+harambasicluka@users.noreply.github.com>

AlexanderAlferov committed:
Test flavour build with extract native libs flag set to true
Test pipeline config

AlexanderAlferov committed:
Merge remote-tracking branch 'origin/release/1.7.x' into feature/3325-pipeline-build-to-tsystems

AlexanderAlferov committed:
Config clean up

d4rken committed:
Dependency injection for workers (DEV) (#1503)

* Allow WorkerManager's workers to be injected.

* Remove static access for Playbook

* Add mock for playbook so dagger can create the worker factory map.

d4rken committed:
Introduce skeleton classes for UI tests and screenshots (EXPOSUREAPP-2997) (#1496)

* Initial refactoring on the Submission flow.
Trimming down SubmissionViewModel.kt

* Refactored submission tan, and began instrumentation tests. Intro, Dispatcher and Tan complete

* Adapted instrumentation tests after merge - added ToDos

* Completed boilerplate submission instrumentation tests. Verification and screenshots to follow

* Fix lints

* More lint issues.

* Fix instrumentation test dependency missmatch.

* Move screenshot related permissions into a debug AndroidManifest.xml,
such that they don't get included in production builds.

* Added missing new line to manifest

Co-authored-by: Oliver Zimmerman <oezimmerman@gmail.com>
Co-authored-by: Kolya Opahle <k.opahle@sap.com>
Co-authored-by: harambasicluka <64483219+harambasicluka@users.noreply.github.com>

harambasicluka committed:
Merge branch 'release/1.7.x' into feature/3325-pipeline-build-to-tsystems

AlexanderAlferov committed:
Push RC build to T-Systems (EXPOSUREAPP-3325)(#1504)

Push RC build to T-Systems (EXPOSUREAPP-3325)

ralfgehrer committed:
Introduce new notification title and body (EXPOSUREAPP-3454) (#1517)

AndroidMedaGalaxy committed:
test cases added as new potential guidelines (DO NOT MERGE YET)

chris-cwa committed:
Converted risk level transaction to task (EXPOSUREAPP-2842) (#1412)

* converted risk level transaction to task

* simplified

* fixed config: timeout is now duration

* less is more

* arrangement + empty lines

* Added additional checks to notification of exposure/test-result to check for finished submission

Signed-off-by: Kolya Opahle <k.opahle@sap.com>

* submit risk level task requests

* - unreachable catch blocks

* housekeeping

* satisfied ktlint + detekt

* fixed flow: isRefreshing

* fixed flavour for testers

* refactored benchmark

* fixed package declaration

* unit tests

* fixed injection

* fixed tests

* comments from PR

* fixed injection

* fixed lint issues

* fixed another merge conflict

* - unused class

* map task factory

* fulfilled comment style wishes

* fixed tests suspended by changes from develop branch

* Update Corona-Warn-App/src/main/java/de/rki/coronawarnapp/risk/DefaultRiskLevels.kt

https://github.com/corona-warn-app/cwa-app-android/pull/1412#discussion_r514325776

Co-authored-by: Kolya Opahle <k.opahle@sap.com>

* changes from PR

* fixed a test

* show logs, ktlint issues

* made log tag non-nullable

* improvements from reviewers

* reduced static access

* Saint Pipeline, please give me the green light

* toString()

* ktlint + di

* fixed test

Co-authored-by: Kolya Opahle <k.opahle@sap.com>
Co-authored-by: harambasicluka <64483219+harambasicluka@users.noreply.github.com>

Corona-Warn-App-Technical-User committed:
Protocol Buffer update for release 1.7.x at Fri Oct 30 10:36:51 UTC 2020 (#1506)

* Update from https://github.com/corona-warn-app/cwa-protocol-buffers/commit/a5a77c4be423c272316b3e30206d561d8217e46f

* Update from https://github.com/corona-warn-app/cwa-protocol-buffers/commit/2cb89bb9aad80ce530e03c420942ff5a7e18b8cb

* Update from https://github.com/corona-warn-app/cwa-protocol-buffers/commit/ca2fbdfae1b1a74244659f40b1e95f877ad3432f

Co-authored-by: AlexanderAlferov <64849422+AlexanderAlferov@users.noreply.github.com>
Co-authored-by: Matthias Urhahn <matthias.urhahn@sap.com>
Co-authored-by: Ralf Gehrer <ralfgehrer@users.noreply.github.com>

BMItr committed:
Fix/satisfy pipeline - gradle def function (EXPSOUREAPP-2572) (#1519)

* moved getHash call to DeviceForTesters Flavor to satisfy CircleCI

* added variant prefix

chris-cwa committed:
Merge remote-tracking branch 'origin/release/1.6.x' into 1.6_to_1.7

# Conflicts:
#	gradle.properties

chris-cwa committed:
Merge branch 'release/1.7.x' into 1.6_to_1.7

AndroidMedaGalaxy committed:
Merge branch 'release/1.7.x' into feature/QR-Scanner-UPPER-case-GUID

AndroidMedaGalaxy committed:
UPPER CASE QR Code can now be accepted; Test cases added

ralfgehrer committed:
1.6 to 1.7 (DEV) #1521

1.6 to 1.7

d4rken committed:
Show current env when launching app (DEV) (#1515)

* On flavor "deviceForTesters" show a toast to tell the testers/devs which environment is currently activated.

* Add unit tests for new "Current env: ..." toast.

Co-authored-by: Ralf Gehrer <ralfgehrer@users.noreply.github.com>

AndroidMedaGalaxy committed:
Merge branch 'release/1.7.x' into feature/QR-Scanner-UPPER-case-GUID

AndroidMedaGalaxy committed:
test cases for invalid QR code with extra slashes added

AndroidMedaGalaxy committed:
Merge branch 'feature/QR-Scanner-UPPER-case-GUID' of https://github.com/corona-warn-app/cwa-app-android into feature/QR-Scanner-UPPER-case-GUID

mlenkeit commented on Corona-Warn-App/src/test/java/de/rki/coronawarnapp/service/submission/ScanResultTest.kt:
```diff
@@ -49,5 +74,7 @@ class ScanResultTest {
     fun extractGUID() {
         QRScanResult("https://localhost/?$guid").guid shouldBe guid
+        QRScanResult("https://localhost/?$lowerCaseGuid").guid shouldBe lowerCaseGuid
+        QRScanResult("https://localhost/?$mixedCaseGuid").guid shouldBe mixedCaseGuid
```
```suggestion
        QRScanResult("https://localhost/?$mixedCaseGuid").guid shouldBe mixedCaseGuid
        QRScanResult("HTTPS://LOCALHOST/?$guid").guid shouldBe guid
        QRScanResult("HTTPS://LOCALHOST/?$lowerCaseGuid").guid shouldBe lowerCaseGuid
        QRScanResult("HTTPS://LOCALHOST/?$mixedCaseGuid").guid shouldBe mixedCaseGuid
```

mlenkeit replied:
Let's make sure that upper/lower case protocal and hostname don't change the guid.

ritsam replied:
Test cases have been added

AndroidMedaGalaxy committed:
GUID Test cases added

AndroidMedaGalaxy committed:
additional test cases to extract guid

additional test cases to check upper/lower case protocal and hostname don't change the guid.

mlenkeit reviewed (approved):
LGTM 👍

harambasicluka reviewed (changes requested):
This feature is for 1.6.0. Please adjust it accordingly, I changed the base branch to release/1.6.x.

mlenkeit commented:
I think this PR needs to be rebased after Luka changed the base branch...

harambasicluka commented:
Small remark, nothing to change now: the ticket is a bug, according to our internal wiki that means the branch has to start with `fix/` and also the ticket number should be included: `fix/3546-qr-scanner-upper-case-guid`.

harambasicluka commented:
Close this as there where some problems with the base branch, new PR can be found here #1524 .
